Issue 249

hlim1 opened the pull request:
Handled the issue of function AST does not get printed by rectify.py.
Handled the issue of pyTranslate.py does not print the function properly.
Tested with the given example.

`    integer function foo(x)
      implicit none
      integer x
      foo = x + 1234567      ! return value of the function
      end function foo

      program main
      integer u, foo
      u = foo(3)
 10   format(I8)
      write (*,10) u
      end program main`

skdebray commented on delphi/translators/for2py/pyTranslate.py:
```diff
@@ -703,7 +706,28 @@ def printFuncReturn(self, node, printState: PrintState):
+                        "right" in node
+                    ), f"Something is missing. Detail of node content: {node}"
+
+                    left = ""
```
This assignment seems unnecessary, since the variable is overwritten in the if statement.

skdebray commented on delphi/translators/for2py/pyTranslate.py:
```diff
@@ -703,7 +706,28 @@ def printFuncReturn(self, node, printState: PrintState):
+                    else:
+                        left = node["left"][0]["value"]
+                    operator = node["operator"]
+                    right = ""
```
This assignment seems unnecessary, since the variable is overwritten in the if statement.

skdebray commented on delphi/translators/for2py/pyTranslate.py:
```diff
@@ -181,11 +181,13 @@ def __init__(self):
         # names
         self.nameMapper = {}
         # Dictionary to hold functions and its arguments
+        self.functions = []
```
Please add a comment to each of these fields saying what it's for.  Also, please elaborate on the comment before `nameMapper`, which currently is not very helpful.

hlim1 commented:
@skdebray As requested, I've removed the unnecessary left and right variables. It was my habit of declaring before use from C/C++, which is not needed in Python. I should look into other codes whether I'm doing the same thing, and remove them as necessary. Also, I added comments for all declared member variables, lists, and dictionaries. I hope they make more sense now.
Thank you.
